crates/iceberg-catalog/src/service/hooks.rs

```diff
@@ -0,0 +1,280 @@
+            hook.register_table(
+                warehouse_id,
+                parameters.clone(),
+                request.clone(),
+                table_ident_uuid,
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+
+    async fn create_view(
+        &self,
+        warehouse_id: WarehouseIdent,
+        view_ident_uuid: ViewIdentUuid,
+        parameters: NamespaceParameters,
+        request: Arc<CreateViewRequest>,
+        data_access: DataAccess,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        for hook in &self.0 {
+            hook.create_view(
+                warehouse_id,
+                view_ident_uuid,
+                parameters.clone(),
+                request.clone(),
+                data_access,
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+
+    async fn commit_view(
+        &self,
+        warehouse_id: WarehouseIdent,
+        view_ident_uuid: ViewIdentUuid,
+        parameters: ViewParameters,
+        request: Arc<CommitViewRequest>,
+        data_access: DataAccess,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        for hook in &self.0 {
+            hook.commit_view(
+                warehouse_id,
+                view_ident_uuid,
+                parameters.clone(),
+                request.clone(),
+                data_access,
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+
+    async fn drop_view(
+        &self,
+        warehouse_id: WarehouseIdent,
+        parameters: ViewParameters,
+        drop_params: DropParams,
+        view_ident_uuid: ViewIdentUuid,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        for hook in &self.0 {
+            hook.drop_view(
+                warehouse_id,
+                parameters.clone(),
+                drop_params.clone(),
+                view_ident_uuid,
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+
+    async fn rename_view(
+        &self,
+        warehouse_id: WarehouseIdent,
+        prefix: Option<Prefix>,
+        view_ident_uuid: ViewIdentUuid,
+        request: Arc<RenameTableRequest>,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        for hook in &self.0 {
+            hook.rename_view(
+                warehouse_id,
+                prefix.clone(),
+                view_ident_uuid,
+                request.clone(),
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+}
+
+#[async_trait::async_trait]
+pub trait EndpointHooks: Send + Sync {
+    async fn commit_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _prefix: Option<Prefix>,
+        _request: Arc<CommitTransactionRequest>,
+        _table_ident_map: Arc<HashMap<TableIdent, TableIdentUuid>>,
+        _request_metadata: Arc<RequestMetadata>,
```

Comment:
We should also add the actual resulting metadata - as this contains information required for the decision if for example snapshot expiration is required.
Also for all other endpoints where we have it in memory anyway.